sql/core/src/main/scala/org/apache/spark/sql/execution/CacheManager.scala

```diff
@@ -72,12 +71,12 @@ class CacheManager extends Logging {
 
   /** Clears all cached tables. */
   def clearCache(): Unit = writeLock {
-    cachedData.asScala.foreach(_.cachedRepresentation.cacheBuilder.clearCache())
-    cachedData.clear()
+    cachedData.foreach(_.cachedRepresentation.cacheBuilder.clearCache())
+    cachedData = IndexedSeq[CachedData]()
   }
 
   /** Checks if the cache is empty. */
-  def isEmpty: Boolean = readLock {
```

Comment:
can we remove the `readLock` method?

Reply:
I think we can remove `writeLock` as well, and simply use `this.synchronized`

Reply:
I made this change, but while doing that I realized there was an issue with the earlier change to the "partition" function.  See comment below

   val (plansToUncache, remainingPlans) = cachedData.partition(cd => shouldRemove(cd.plan))
   // If a new plan is cached by a different thread at this point, it will be in the cachedData object,
   //  but not in plansToUncache or remainingPlans.  So the next line will remove it.
    writeLock {
      cachedData = remainingPlans
    }

Reply:
So I reverted this back to the previous behavior.